Expanded Data Type docs

MoltonMontro opened the pull request:
- Split **flag**, **vector3**, and **color** into their own docs.
- **ELightingVision** enum doc.
- Revised older enum/struct docs to use tables, and nested these docs into their own folders.
- Listed relevant **C# built-in types**, with common aliases and default values.
- Updated TOC/index.

----

This PR is ready (only contains commits related to data types), just a few questions below.

1. Leave objects/arrays in Data File Format doc? I _think_ it makes sense to leave them there, due to how they're used when modding.
2. "Named Value" column name good (for enums)? My concern was making sure readers knew how these values related to the enum. Can be changed to "Name" instead if you're confident.
3. I've nested all Enum docs into **/enum**, and PlayerSpotLightConfig into **/struct**.
4. **C# built-in types** used to have a column for example values (on an earlier draft PR), but I've replaced this with the type's range. This seems more useful.
5. The doc for C# built-in types includes a column for common name, and a column for alias(es). I've sorted these out based on how Microsoft treats them, but a lot of the properties in our documentation currently uses terms from the "aliases" column. Do we have a preference here? E.g., should we replace all instances of "byte" with "uint8", or instances of "uint64" with "ulong", or "float" with "single"...

MoltonMontro committed:
Data types

- Split flag, vector3, and color into their own docs.
- ELightingVision enum doc.
- Revised older enum/struct docs to use tables.
- C# built-in types, with common aliases and default values.
- Updated TOC.

SDGNelson commented:
Looking good, thanks!

1-4 all make sense to me.

5. For floating point numbers I think we should use `float32` and `float64`. For integers `int32` and `uint32` style. The older naming is more confusing when you're just trying to convey that a value has e.g. a 16-bit or 64-bit range. At least c# `ulong` isn't as bad as c++ with `unsigned long long int`.

MoltonMontro committed:
Reorganized type names

For floats we'll use float32 (etc.), for integers we'll use int32 and uint32 (etc.).

Added double/float64 as well, although I don't think any docs reference it.

MoltonMontro committed:
Markdown fixes

MoltonMontro commented:
I've moved the type names around per the above [& added float64/double to the table]. :)

We can update the terms actually used on other pages in future PRs. Most of them will get updated when converting asset property lists into tables / adding Unity Setup info.

SDGNelson commented:
Alright: I'll merge now then!
